plugins/dell-dock2/fu-dell-dock2-ec.c

```diff
@@ -0,0 +1,972 @@
+/*
+ * Copyright 2024 Dell Technologies
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR MIT
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-dell-dock2-common.h"
+
+#define BIT_SET(x, y)	(x |= (1 << y))
+#define BIT_CLEAR(x, y) (x &= (~(1 << y)))
+
+const FuHIDI2CParameters dell_dock2_ec_base_settings = {
+    .i2ctargetaddr = DELL_DOCK2_EC_HID_I2C_ADDRESS,
+    .regaddrlen = 1,
+    .i2cspeed = I2C_SPEED_250K | 0x80,
+};
+
+typedef struct __attribute__((packed)) {
+	guint32 ec_version;
+	guint32 mst_version;
+	guint32 hub1_version;
+	guint32 hub2_version;
+	guint32 tbt_version;
+	guint32 pkg_version;
+	guint32 pd_version;
+	guint32 epr_version;
+	guint32 dpmux_version;
+	guint32 rmm_version;
+	guint32 reserved[6];
+} FuDellDock2DockFWVersion;
+
+typedef struct __attribute__((packed)) {
+	guint8 dock_configuration;
+	guint8 dock_type;
+	guint16 power_supply_wattage;
+	guint16 module_type;
+	guint16 board_id;
+	guint16 port0_dock_status;
+	guint16 port1_dock_status; // unused for K2 dock, should be 0
+	guint32 dock_firmware_pkg_ver;
+	guint64 module_serial;
+	guint64 original_module_serial;
+	guint8 service_tag[7];
+	guint8 marketing_name[32];
+	guint32 dock_error;
+	guint32 dock_module_status;
+	guint32 dock_module_error;
+	guint8 reserved;
+	guint32 dock_status;
+	guint16 dock_state;
+	guint16 dock_config;
+	guint8 dock_mac_addr[48];
+	guint32 dock_capabilities;
+	guint32 dock_policy;
+	guint32 dock_temperature;
+	guint32 dock_fan_speed;
+	guint8 unused[35];
+} FuDellDock2DockDataStructure;
+
+typedef struct __attribute__((packed)) {
+	struct FuDellDock2V2DockInfoHeader {
+		guint8 total_devices;
+		guint8 first_index;
+		guint8 last_index;
+	} header;
+	struct FuDellDock2EcQueryEntry {
+		struct FuDellDock2V2EcAddrMap {
+			guint8 location;
+			guint8 device_type;
+			guint8 sub_type;
+			guint8 arg;
+			guint8 instance;
+		} ec_addr_map;
+		union {
+			guint32 version_32;
+			guint8 version_8[4];
+		} __attribute__((packed)) version;
+	} devices[20];
+} FuDellDock2DockInfoStructure;
+
+/* Private structure */
+struct _FuDellDock2Ec {
+	FuDevice parent_instance;
+	FuDellDock2DockDataStructure *dock_data;
+	FuDellDock2DockInfoStructure *dock_info;
+	FuDellDock2DockFWVersion *raw_versions;
+	FuDellDock2BaseType base_type;
+	guint8 base_sku;
+	gchar *ec_minimum_version;
```

Comment:
I don't see you ever using this variable, I think it can go.

Reply:
Fixed, thanks.